Composer/packages/client/src/components/ProjectTree/ExpandableNode.tsx

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/** @jsx jsx */
+import { jsx, css } from '@emotion/core';
+import { useState, MouseEvent, KeyboardEvent } from 'react';
+
+type Props = {
+  children: JSX.Element;
+  summary: JSX.Element;
+  depth?: number;
+  detailsRef?: (el: HTMLElement | null) => void;
+};
+
+const summaryStyle = css`
+  label: summary;
+  display: flex;
+  padding-left: 12px;
+  padding-top: 6px;
+`;
+
+const nodeStyle = (depth: number) => css`
+  margin-left: ${depth * 16}px;
+`;
+
+export const ExpandableNode = ({ children, summary, detailsRef, depth = 0 }: Props) => {
+  const [isOpen, setOpen] = useState(true);
+
+  function handleClick(ev: MouseEvent) {
+    if ((ev.target as Element)?.tagName.toLowerCase() === 'summary') {
+      setOpen(!isOpen);
+    }
+    ev.preventDefault();
+  }
+
+  function handleKey(ev: KeyboardEvent) {
+    if (ev.key === 'Enter' || ev.key === 'Space') setOpen(!isOpen);
```

Comment:
Maybe use KeyCodes typescript interface?

ev.keyCode === Keycodes.Enter?

Reply:
keyCodes are deprecated, recommended is the way Ben has it:
https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent

Reply:
I didn't realize that. I had always been using keycodes instead of the string. Good to know :)